projects/angular/src/data/datagrid/datagrid-row.spec.ts

```diff
@@ -543,6 +543,123 @@ export default function (): void {
         context.detectChanges();
       }
     });
+
+    describe('Details', function () {
+      let context: TestContext<ClrDatagridRow<Item>, DatagridWithDisabledOrHiddenDetails>;
+
+      beforeEach(function () {
+        context = this.create(ClrDatagridRow, DatagridWithDisabledOrHiddenDetails, DATAGRID_SPEC_PROVIDERS);
+        context.detectChanges();
+      });
+
+      it('displays a clickable double caret when the row have details', function () {
+        expect(context.clarityElement.querySelector('button cds-icon[shape^=angle-double]')).not.toBeNull();
+      });
+
+      it('Detail button must contain aria-controls', function () {
+        const button = context.clarityElement.querySelector('button.datagrid-detail-caret-button');
+        expect(button).not.toBeNull();
+        expect(button.getAttribute('aria-controls')).not.toBeNull();
+      });
+
+      it('display the details when pre selected', fakeAsync(function () {
+        context.testComponent.preState = context.testComponent.items[0];
+        tick();
+        context.detectChanges();
+
+        const button = context.clarityElement.querySelector('button.datagrid-detail-caret-button');
+        const detailBody = document.querySelector('div.clr-dg-detail-body-wrapper') as HTMLElement;
+
+        expect(button).not.toBeNull();
+        expect(button.querySelector('cds-icon[shape^=angle-double][direction^=left]')).not.toBeNull();
+        expect(detailBody.innerText).toBe('' + context.testComponent.preState.id);
```

Comment:
Do we need to check for the button and icon again if we have that in the test above?

Reply:
Corrected in commit 00e868d